src/states.jl

```diff
@@ -1,22 +1,13 @@
 import Base: ==, +, -, *, /, length, copy, eltype
 import LinearAlgebra: norm, normalize, normalize!
-
-"""
-Abstract base class for [`Bra`](@ref) and [`Ket`](@ref) states.
-
-The state vector class stores the coefficients of an abstract state
-in respect to a certain basis. These coefficients are stored in the
-`data` field and the basis is defined in the `basis`
-field.
-"""
-abstract type StateVector{B,T} end
+import QuantumInterface: StateVector, AbstractKet, AbstractBra
```

Comment:
Two new abstract types were created as well. Now we have `AbstractKet <: StateVector` and `AbstractBra <: StateVector` also for the benefit of `QSymbolics`. Very few changes related to these new abstract types were performed, all in this file